src/core/utils/channel_manager.cpp

```diff
@@ -66,30 +70,70 @@ ChannelManager::ChannelManager(Instance &aInstance)
 
 void ChannelManager::RequestChannelChange(uint8_t aChannel)
```

Comment:
How about if we define a new method like `ChangeCslChannel()` that deals with the CSL channel change (when fetuare is enabled)? and we can leave the `RequestCannelChange()` as before?

Reply:
I split this method, which improves readability. Former `RequestChannelChange()` was renamed to `ReuqestNetworkChannelChange()`. `RequestChannelChange()` provides the common interface for both cases now, and also adds checks for FTD downgraded to SSED.

Reply:
Thanks. LGTM.